packages/flutter/lib/src/material/chip.dart

```diff
@@ -1793,7 +1794,7 @@ class _RawChipState extends State<RawChip> with TickerProviderStateMixin<RawChip
         child: AnimatedBuilder(
           animation: Listenable.merge(<Listenable>[selectController, enableController]),
           builder: (BuildContext context, Widget child) {
-            return Container(
+            return Ink(
               decoration: ShapeDecoration(
                 shape: shape,
```

Comment:
In the before/after pictures the splash doesn't appear to be clipped by the Chip's shape (noticeable on the chip's right side). Presumably this is a pre-existing condition. If that's the case, please file a new issue.

Reply:
That is a pre-existing condition, I filed the issue: 
https://github.com/flutter/flutter/issues/43215